role: add option for keycloak auth and update registry cleanup script

markoburcul opened the pull request:
* add option to enable keycloak auth
* update cleanup script for purging old docker images, set default retention to 180 days
* create separate token for package replication

Referenced issue:
* https://github.com/status-im/infra-ci/issues/121

markoburcul committed:
oauth: add option for enabling keycloak oauth

Referenced issue:
* https://github.com/status-im/infra-ci/issues/121

Signed-off-by: markoburcul <marko@status.im>

jakubgs commented on templates/gitea.ini.j2:
```diff
@@ -41,7 +41,7 @@ ISSUE_INDEXER_PATH = /data/gitea/indexers/issues.bleve
 PROVIDER = db
 COOKIE_SECURE = true
-SAME_SITE = strict
+SAME_SITE = lax
```
Why was this change necessary? Might be worth a comment.

markoburcul replied:
It was necessary for Keycloak OAuth to work since without it I was getting 500 with no specific error when being redirected from Keycloak back to Gitea URL. Here is the [issue](https://github.com/go-gitea/gitea/issues/27033). I can put a comment in the file.

jakubgs replied:
Nice, thanks.

jakubgs commented on tasks/main.yml:
```diff
@@ -11,8 +11,6 @@
   tags: ['role::gitea', 'role::gitea:backup']
 - import_tasks: consul.yml
   tags: ['role::gitea', 'role::gitea:consul']
-- import_tasks: token.yml
-  tags: ['role::gitea', 'role::gitea:token']
```
Why?

markoburcul replied:
Why do we have it two times imported?

jakubgs replied:
Ah, i'm blind.

jakubgs commented on defaults/main.yml:
```diff
@@ -17,10 +17,33 @@ gitea_app_log_level: 'info'
+  "read:admin",
+  "write:admin",
+  "read:organization",
+  "write:organization",
```
```suggestion
gitea_app_admin_user_token_scopes:
  - 'read:admin'
  - 'write:admin'
  - 'read:organization'
  - 'write:organization'
```

jakubgs commented on tasks/token.yml:
```diff
@@ -36,3 +38,32 @@
     gitea_app_admin_token: '{{ gitea_app_admin_token_raw.content | b64decode | trim }}'
   when: gitea_token_file.stat.exists
+
+- name: Check if admin package replication API token file exists
+  stat:
+    path: '{{ gitea_service_path }}/replication-token'
+  register: gitea_replication_token_file
```
Looks like we're just repeating the same stuff as above. Why can't we just re-use `tasks/token.yml` with `include_tasks` but just provide additional `vars` to create a different token?

jakubgs reviewed (approved):
Mostly good.

siddarthkay commented on files/cleanup-registry.py:
```diff
@@ -65,27 +66,28 @@ def get_organization(giteaApi):
+    logging.info(f'Deleting artifact {art.name} of {art.owner}')
     try:
-        res=giteaApi._del('packages/%s/container/%s/%s' % (art.owner, art.name, art.version))
+        path = f'packages/{art.owner}/container/{quote(art.name, safe='')}/{art.version}'
```
```suggestion
        path = f'packages/{art.owner}/container/{quote(art.name, safe="")}/{art.version}'
```

This quote can cause issues like : 

```
Mar 11 00:00:05 node-01.he-eu-fsn1.ci.gitea systemd[1]: Starting Timer to perform clean...
Mar 11 00:00:05 node-01.he-eu-fsn1.ci.gitea cleanup-registry.py[1866291]:   File "/docker/gitea/cleanup-registry.py", line 87
Mar 11 00:00:05 node-01.he-eu-fsn1.ci.gitea cleanup-registry.py[1866291]:     path = f'packages/{art.owner}/container/{quote(art.name, safe='')}/{art.version}'
```

markoburcul committed:
packages: add new token and update cleanup script

Use separate token for package(container images) replication and update
the cleanup script to reflect new gitea api changes. Increase retention
for images to default 180 days.

Referenced issue:
* https://github.com/status-im/infra-ci/issues/121

Signed-off-by: markoburcul <marko@status.im>

jakubgs reviewed (approved):
Looks good. Though in the future if we need more tokens we should probably just call `token.yml` in a loop over some list.
